core/domain/user_services.py

```diff
@@ -499,6 +501,161 @@ def get_user_settings_by_auth_id(
         return None
 
 
+def get_all_user_groups() -> List[user_domain.UserGroup]:
+    """Return the list of user groups.
+
+    Returns:
+        List[user_domain.UserGroup]. List of all user groups.
+    """
+    user_group_models: List[user_models.UserGroupModel] = list(
+        user_models.UserGroupModel.get_all())
+    user_groups_list = []
+    for user_group_model in user_group_models:
+        user_usernames: List[str] = []
+        for user_setting in get_users_settings(user_group_model.user_ids):
+            assert user_setting is not None
```

Comment:
Do you need some sort of custom error message to say which username is invalid, so that whoever is debugging knows what exactly needs to be investigated? I think with assert statements in Python you can add an optional error message.

Ditto below.

Reply:
I see, I have added the custom message, we will be able to see the user_id of the user whose username is not valid.